main.go

```diff
@@ -145,29 +146,11 @@ func main() {
 		go graphite.Graphite(metrics.DefaultRegistry, 1*time.Second, *metricsPrefix, *graphiteAddr)
 	}
 
-	hostname, err := os.Hostname()
-	if err != nil {
-		fmt.Fprintf(os.Stderr, "error: unable to get hostname: %s", err)
-		os.Exit(1)
-	}
-
-	metrics := &metricsConfig{
-		url:      *metricsURL,
-		registry: metrics.DefaultRegistry,
-		prefix:   *metricsPrefix,
-		hostname: hostname,
+	if *metricsURL != "" {
+		logger.Printf("metrics enabled; reporting metrics via POST to %s", *metricsURL)
 	}
 
-	if metrics.url != "" {
-		logger.Printf("metrics enabled; reporting metrics via POST to %s", metrics.url)
-		go metrics.publishMetrics()
-	}
-
-	if metrics.url != "" || *graphiteAddr != nil || *statusAddr != nil {
-		// We only bother collecting memory usage metrics if we're
-		// actually configured to expose them somewhere.
-		go metrics.collectSystemMetrics()
-	}
+	metrics := sqmetrics.NewMetrics(*metricsURL, *metricsPrefix, metrics.DefaultRegistry)
```

Comment:
The `metrics` var gets put into the context, so I can grab it later down the line to serve it through the status port (if the status port is enabled).